dynamite_flask_app/create_update_database.py

```diff
@@ -0,0 +1,188 @@
+"""This module creates and updates the database tables."""
+
+import json
+import random
+
+from init_db import db
+from models import User, Product, Order, ImageUrl, Feature, ProductSize, Size, Category, Feature, ActiveToken
+from resources.user import add_new_user
+
+
+with open('products.json') as file:
+    """Load the products.json file into the products variable."""
+    products = json.load(file)
+
+users = [
+    {
+        'name': 'Zain',
+        'email': 'abc@123.com',
+        'password': 'asd123',
+        'address': 'abc',
+        'contact_no': 123,
+    },
+    {
+        'name': 'Ali',
+        'email': 'abc1@123.com',
+        'password': 'asd123',
+        'address': 'ab2c',
+        'contact_no': 1523,
+    },
+    {
+        'name': 'Shah',
+        'email': 'abcfff1@123.com',
+        'password': 'asd123',
+        'address': 'ab2cs',
+        'contact_no': 12345523,
+    },
+    {
+        'name': 'Ahmed',
+        'email': 'abhhewc1@123.com',
+        'password': 'asd123',
+        'address': 'ab2erc',
+        'contact_no': 176474523,
+    },
+    {
+        'name': 'Boy',
+        'email': 'furrfurr@123.com',
+        'password': 'asd123',
+        'address': 'fwefeew',
+        'contact_no': 8484876,
+    }
+]
+
+orders = [
+    {
+        'date': '2008, 4, 12'
+    },
+    {
+        'date': '2018, 12, 02'
+    }
+]
+
+
+def add_products():
+    """Adds all products with their details to the database."""
+    for product in products:
+        product_details = Product(
+            product_id=product['id'], name=product['name'], price=product['price'], sale_price=product['sale_price'],
+            description=product['description']
+        )
+
+        add_image_url(product, product_details)
+        add_feature(product, product_details)
+        add_category(product_details)
+        add_size(product, product_details)
+
+        db.session.add(product_details)
+
+    db.session.commit()
+
+
+def add_image_url(product, product_details):
+    """Adds image urls to the database.
+
+    Args:
+        Product(str): Each product, used from the add_products() method.
+        product_details: Add product to the database, also used to create relationship with the Product table.
+
+    """
+    for image_url in product['img_urls']:
+        url = ImageUrl(url=image_url)
+        product_details.img_urls.append(url)
+
+
+def add_feature(product, product_details):
+    """adds features to the database.
```

Comment:
This comment does not anything. comments should either explain some business logic or should be omitted altogether.

Reply:
done